internal/hwmon/hwmon.go

```diff
@@ -31,8 +31,8 @@ type HwMonController struct {
 	Platform string
 	Path     string
 
-	// Fans maps from HwMon index -> HwMonFan instance
-	Fans map[int]*fans.HwMonFan
+	// Fans (can be matched either by enumeration index or channel number)
+	Fans []fans.HwMonFan
```

Comment:
Hmm isn't this problematic if there is a gap in the channel numbering?
What if there only exist the fan channels 1, 3 and 5? Is this unreasonable?
I thought the whole point of this was (despite actually using the channel instead of indexing) to allow arbitrary channel numbers (which might not be aligned).

Reply:
I switched to a slice because a fan can be matched either by enumeration index or by channel number, so there isn't a unique key to do lookups. I do not index into this slice, I iterate over it. So gaps in channel numbers are fine. Perhaps the comment can mention this. There is basically no performance problem with iterating over a small slice vs a map.

Reply:
Ah ok, so you always iterate and the matching is done using the `if` statements I mentioned above? Makes sense 👍